framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -923,6 +1066,50 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
+        if (range.from > range.to || range.from == Long.MIN_VALUE && range.to == Long.MAX_VALUE) {
+            return UNKNOWNVAL;
+        } else if (!range.isWiderThan(MAX_VALUES)) {
```

Comment:
I don't see the benefit of converting a range to an `@IntVal`.
 * All the routines like `isSubtype` and `greatestlowerBound` will handle both.
 * Converting will make the error messages state a different type than what the user wrote.
 * Converting may make the error messages more verbose, if the range has many values (around 10).
I agree that converting to `@IntVal` would be clearer if there are only a small number of possible values, but I don't see a compelling need for that.